### ISSUE TITLE
Private instance method throws error, Cannot read property 'flags' of undefined

### ISSUE THREAD

**Issue by jperl** [palantir/tslint#1601]
### Bug Report
- **TSLint version**: `3.15.1` and `4.0.0-dev.0`
- **TypeScript version**: `2.0.3`
- **Running TSLint via**: CLI
#### TypeScript code being linted

``` ts
 private getQuality = ({ dataKey, rowData: { quality } }) => quality[dataKey];
```
#### Actual behavior

```
/Users/jperl/dev/my-project/node_modules/typescript/lib/typescript.js:49932
            var flags = symbol.flags;
                              ^
TypeError: Cannot read property 'flags' of undefined
    at displayPartKind (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:49932:31)
    at symbolPart (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:49930:34)
    at Object.writeSymbol (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:49917:31)
    at buildBindingElementDisplay (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18782:28)
    at /Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18763:110
    at buildDisplayForCommaSeparatedList (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18809:21)
    at buildBindingPatternDisplay (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18763:21)
    at buildParameterDisplay (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18747:21)
    at buildDisplayForParametersAndDelimiters (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18837:21)
    at buildSignatureDisplay (/Users/jperl/my-project/node_modules/typescript/lib/typescript.js:18883:17)
```
#### Expected behavior

No error.

**Comment by adidahiya** [comment]
what does your tslint.json look like? can you track down which rule is throwing this exception? that stack trace doesn't have any reference to tslint in it, which is a bit strange...

**Comment by jperl** [comment]
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    "no-console": [false],
    "quotemark": [true, "single"]
  }
}

**Comment by jperl** [comment]
When I change the code to 

```
  private getQuality = (options) => {
    const { dataKey, rowData: { quality } } = options;
    return quality[dataKey];
  };
```

it runs fine

**Comment by grovesNL** [comment]
I just ran into the same issue, i.e.:

```
function test({ x: { y } }) {
    console.log(y);
}
test({ x: { y: "test" } });
```

will fail with the `var flags = symbol.flags` exception.

Here's the complete call stack while linting the above example:
![tslint](https://cloud.githubusercontent.com/assets/2113872/19536340/d1a4bfd4-9608-11e6-81d9-b99e39497ca4.png)

Perhaps a TypeScript issue?

**Comment by adidahiya** [comment]
looks like a duplicate of #1641, see my comment: https://github.com/palantir/tslint/issues/1641#issuecomment-254884450

I recommend disabling the no-unused-variable rule and using the compiler checks instead

**Comment by grovesNL** [comment]
@adidahiya Thanks, I missed that issue. Will do!

**Comment by andy-ms** [comment]
Should be fixed by #2235.
